### PR TITLE
feat(stock): add calculateRealCosts with real consumption analysis and 6m projection

### PR DESCRIPTION
**Pull request by coelhotv**
## Summary
- Added `calculateRealCosts()` function to analyze medication costs using **real consumption data** from logs
- Falls back to **theoretical consumption** if < 14 days of data
- Returns **projection for 6 months** (in addition to 3m projection)
- Updated CostChart component with **real/estimated badge**
- Comprehensive test coverage for all scenarios

## Changes
- `src/features/stock/services/costAnalysisService.js`: New `calculateRealCosts()` function (130 lines)
  - Analyzes real vs. theoretical consumption
  - Detects when >= 14 days of dose logs available
  - Returns `isRealData` flags at item and response level
  - Calculates both 3m and 6m projections
- `src/features/stock/components/CostChart.jsx`: Updated component (9 lines)
  - Added `isRealData` and `projection6m` props
  - Displays badge: "Consumo real" (green) or "Estimativa" (purple)
  - Shows both 3m and 6m projections side-by-side
- `src/features/stock/components/CostChart.css`: New styles (40 lines)
  - Badge styling with appropriate colors
  - Responsive grid for projections (2 columns → 1 on mobile < 360px)
- `src/features/stock/services/__tests__/costAnalysisService.test.js`: Extended with 10 new test cases
  - Happy path: >= 14 days of data uses real consumption
  - Fallback: < 14 days uses theoretical consumption
  - Threshold behavior at exactly 14 days
  - Projection calculations (3m and 6m)
  - Edge cases: null inputs, no data, missing prices

## Quality Gate
- ✅ 502/502 tests passing (all critical services)
- ✅ 0 lint errors
- ✅ Build succeeds
- ✅ 90%+ coverage on new code
- ✅ No Supabase calls in service (pure function)

## Test Plan
1. View Stock tab with 14+ days of dose logs
   - Verify badge shows "Consumo real" (green)
   - Verify daily consumption matches actual intake
   - Verify 3m and 6m projections show

2. View Stock tab with < 14 days of data
   - Verify badge shows "Estimativa" (purple)
   - Verify daily consumption falls back to theoretical
   - Verify projections calculated correctly

3. Add more dose logs in next days
   - Verify badge transitions from "Estimativa" to "Consumo real" after reaching 14 days
   - Verify cost projections update to reflect actual consumption

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by coelhotv** [comment]
🤖 **Gemini Code Assist** criou issue #310 para tracking desta sugestão de refactoring.

File: src/schemas/costAnalysisSchema.js [hunk 1]
```diff
@@ -57,3 +57,23 @@ export const CalculateDailyIntakeInputSchema = z.object({
...
+    .min(0, 'medicines deve ser um array')
+    .optional()
+    .default([]),
+  protocols: z.array(ProtocolSchema).min(0, 'protocols deve ser um array').optional().default([]),
+  logs: z
+    .array(
+      z.object({
+        medicine_id: z.string().min(1, 'medicine_id é obrigatório'),
+        taken_at: z.string().or(z.date()),
+        quantity_taken: z.number().nonnegative('quantity_taken deve ser >= 0').optional().default(0),
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply src/schemas/costAnalysisSchema.js]: sugestão aplicada no commit 1569b75
@gemini-code-assist pode validar

**Comment by coelhotv** [comment]
🤖 **Gemini Code Assist** criou issue #311 para tracking desta sugestão de refactoring.

File: src/features/stock/services/costAnalysisService.js [hunk 2]
```diff
@@ -157,6 +159,112 @@ export function calculateProjection(monthlyCost, months = 3) {
...
+
+      if (daysWithData >= 14) {
+        // Consumo real: total de comprimidos consumidos / dias com dados
+        const totalConsumed = medLogs.reduce((sum, l) => sum + (l.quantity_taken || 0), 0)
+        dailyConsumption = daysWithData > 0 ? totalConsumed / daysWithData : 0
+        isRealData = true
+      } else {
+        // Fallback: consumo teórico baseado no protocolo
+        dailyConsumption = calculateDailyIntake(med.id, validatedProtocols)
+        isRealData = false
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply src/features/stock/services/costAnalysisService.js]: sugestão aplicada no commit 6fe979f
@gemini-code-assist pode validar

**Comment by coelhotv** [comment]
🤖 **Gemini Code Assist** criou issue #314 para tracking desta sugestão de refactoring.
